File: java/src/main/java/com/dokmatiq/docgen/model/WatermarkConfig.java
package com.dokmatiq.docgen.model;

import com.fasterxml.jackson.annotation.JsonInclude;

/** Diagonal watermark overlay configuration. */
@JsonInclude(JsonInclude.Include.NON_NULL)
public record WatermarkConfig(
        String text,
        Double fontSize,
        Double opacity,
        String color
) {
    public WatermarkConfig(String text) {
        this(text, null, null, null);
    }
}

File: php/src/Model/WatermarkConfig.php
<?php

declare(strict_types=1);

namespace Dokmatiq\DocGen\Model;

use Dokmatiq\DocGen\Internal\Serializer;

/** Diagonal watermark overlay configuration. */
final class WatermarkConfig
{
    public function __construct(
        public readonly string $text,
        public readonly ?float $fontSize = null,
        public readonly ?float $opacity = null,
        public readonly ?string $color = null,
    ) {}

    /** @return array<string, mixed> */
    public function toArray(): array
    {
        return Serializer::toArray($this);
    }
}
